mesonbuild/backend/ninjabackend.py

```diff
@@ -1908,11 +1908,11 @@ def generate_rust_target(self, target: build.BuildTarget) -> None:
                 # specify `extern CRATE_NAME=OUTPUT_FILE` for each Rust
                 # dependency, so that collisions with libraries in rustc's
                 # sysroot don't cause ambiguity
-                #
-                # Also convert crate names with dashes to underscores like
+
+                # Convert crate names with dashes to underscores by default like
                 # cargo does as dashes can't be used as parts of identifiers
                 # in Rust
-                d_name = d.name.replace('-', '_')
+                d_name = target.rust_dependency_map.get(d.name) or d.name.replace('-', '_')
```

Comment:
This code with the comment above is duplicated in 2 places, maybe worth factoring that into `target.get_rust_dependency_name(d)`?